### PR TITLE
Allow `MottonenStatePreparation` to be JIT compiled with JAX

### PR DESCRIPTION
**Pull request by eddddddy**
**Context:**
Previously the `MottonenStatePreparation` template cannot be JIT compiled with JAX. See the GitHub issue below for more information.

**Description of the Change:**
Allow `MottonenStatePreparation` to be JIT compiled.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
Closes https://github.com/PennyLaneAI/pennylane/issues/3231

### CODE REVIEW

File: pennylane/math/single_dispatch.py [hunk 1]
```diff
@@ -201,6 +201,7 @@ def _take_autograd(tensor, indices, axis=None):
 ar.autoray._SUBMODULE_ALIASES["tensorflow", "sinc"] = "tensorflow.experimental.numpy"
 ar.autoray._SUBMODULE_ALIASES["tensorflow", "isclose"] = "tensorflow.experimental.numpy"
 ar.autoray._SUBMODULE_ALIASES["tensorflow", "atleast_1d"] = "tensorflow.experimental.numpy"
+ar.autoray._SUBMODULE_ALIASES["tensorflow", "all"] = "tensorflow.experimental.numpy"
```
Comment (antalszava) [inline pennylane/math/single_dispatch.py:204]: Why was this required?

**Review by antalszava — APPROVED** [review]
Looks good! :+1:

File: pennylane/math/single_dispatch.py (hunk 1 above)
Comment (eddddddy) [reply pennylane/math/single_dispatch.py:204]: It's because `all` is used in one of the validation checks, but `all` doesn't exist as a top-level function in tensorflow. This line tells autoray to look for it in the `tf.experimental.numpy` submodule.
